front/lib/connectors_api.ts

```diff
@@ -238,15 +260,26 @@ async function _resultFromResponse<T>(
       logger.error({ jsonError }, "Unexpected response from ConnectorAPI");
       return new Err(jsonError);
     } else {
-      logger.error(
-        { statusCode: response.status, statusText: response.statusText },
-        "Unexpected response from ConnectorAPI"
-      );
-      return new Err({
-        error: {
-          message: `Unexpected response status: ${response.status} ${response.statusText}`,
-        },
-      });
+      try {
+        const textError = await response.text();
+        const errorResponse = JSON.parse(textError);
+        logger.error(
+          { errorResponse },
+          "Unexpected response from ConnectorAPI"
+        );
+        return new Err(errorResponse);
```

Comment:
I think you want to check that you have a message and a type before returning that object?

Reply:
Otherwise you go in unexpected mode